src/averageAges.js

```diff
@@ -37,7 +42,19 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const mothers = people.map(person => person.mother);
+
+  const women = withChildren === undefined
+    ? people.filter(person => person.sex === 'f')
+    : people.filter(person => (
+      person.sex === 'f' && mothers.includes(person.name)
+    ));
+
+  const averageAge = women.map(person => person.died - person.born)
+    .reduce((a, b) => a + b)
+    / women.length;
```

Comment:
```suggestion
  const averageAge = women
    .map(person => person.died - person.born)
    .reduce((a, b) => a + b)
      / women.length;
```
I think this formatting looks better.